Homeworks/HW_03/hw_03_2.py

```diff
@@ -0,0 +1,53 @@
+       'Пауза между первым и вторым таймами составляет 15 минут, в течение которой команды отдыхают, а по её окончании меняются воротами.' \
+       ' Цель игры - забить мяч в ворота противника, сделать это как можно большее количество раз и постараться не допустить гола в свои ворота.' \
+       'Матч выигрывает команда, забившая большее количество голов. В случае, если в течение двух таймов команды забили одинаковое количество голов,' \
+       ' то или фиксируется ничья, или победитель выявляется согласно установленному регламенту матча. ' \
+       'В этом случае может быть назначено дополнительное время - ещё два тайма по 15 минут каждый. ' \
+       'Как правило, между основным и дополнительным временем матча командам предоставляется перерыв. ' \
+       'Между дополнительными таймами командам даётся лишь время на смену сторон. ' \
+       'Одно время в футболе существовало правило, по которому победителем объявлялась команда, первой забившая гол (правило "золотого гола") ' \
+       'или выигрывавшая по окончании любого из дополнительных таймов (правило "серебряного гола"). ' \
+       'В настоящий момент дополнительное время либо не играется вовсе, либо играется в полном объёме (2 тайма по 15 минут). ' \
+       'Если в течение дополнительного времени победителя выявить не удаётся, проводится серия послематчевых пенальти, не являющихся частью матча: ' \
+       'по воротам противника с расстояния 11 метров пробивается по пять ударов разными игроками. ' \
+       'Если количество забитых пенальти у обеих команд будет равным, тогда пробиваются по одной паре пенальти, пока не будет выявлен победитель. ' \
+       'Матчи могут проводиться как на полях с естественным, так и на полях с искусственным покрытием. ' \
+       'Согласно официальным правилам игры в футбол искусственное покрытие должно быть зелёного цвета. ' \
+       'Поле для игры имеет форму прямоугольника. Боковая линия обязательно должна быть длиннее линии ворот. ' \
+       'Принималось решение о том, что размер поля должен составлять 100-110 м (110-120 ярдов) в длину и минимум 64-75 (70-80 ярдов) в ширину, ' \
+       'однако затем обязательность этого требования была приостановлена.'
+# знаки, исключаемые из текста
+SIGNS = ['.', ',', '"', "'", ';', ':', '!', '?', '-', '(', ')', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+# количество самых частых
+TOP = 10
+# удаляем знаки и приводим к нижнему регистру
+for sign in SIGNS:
+    text = text.replace(sign, '')
+text = text.lower()
+# разбиваем по пробелу
+words_list = text.split()
+# делаем частотный словарь
+freq_dict = {}
+for word in words_list:
+    freq_dict.setdefault(word, 0)
+    freq_dict[word] += 1
+# сортируем по значению, по убыванию
+freq_dict = dict(sorted(freq_dict.items(), key=lambda word: word[1], reverse=True))
+# печатаем первые TOP
+for i, item in enumerate(freq_dict.items(), 1):
+    if i > TOP:
+        break
+    else:
+        print(i, item)
```

Comment:
Можно значительно проще
```suggestion
freq_dict = sorted(freq_dict, key=lambda word: freq_dict [word], reverse=True)[:TOP]
print(freq_dict)
```